src/fw/applib/ui/kino/kino_layer.c

```diff
@@ -16,9 +16,46 @@
 
 #include "kino_layer.h"
 
+#include <stdint.h>  // For uintptr_t
+
 #include "applib/applib_malloc.auto.h"
 #include "applib/graphics/graphics.h"
 
+// Use the LSB of the context pointer as an inversion flag
+// This is a hack because the KinoLayer struct needs to stay the same size
+// due to compatibility reasons
+#define KINO_LAYER_INVERT_FLAG_MASK ((uintptr_t)1)
+#define KINO_LAYER_GET_INVERT_FLAG(kino_layer) \
+  (((uintptr_t)((kino_layer)->context)) & KINO_LAYER_INVERT_FLAG_MASK)
+#define KINO_LAYER_SET_INVERT_FLAG(kino_layer, invert)                                   \
+  do {                                                                                   \
+    (kino_layer)->context =                                                              \
+        (void *)((((uintptr_t)((kino_layer)->context)) & ~KINO_LAYER_INVERT_FLAG_MASK) | \
+                 ((invert) ? KINO_LAYER_INVERT_FLAG_MASK : 0));                          \
+  } while (0)
+#define KINO_LAYER_GET_REAL_CONTEXT(kino_layer) \
+  ((void *)(((uintptr_t)((kino_layer)->context)) & ~KINO_LAYER_INVERT_FLAG_MASK))
```

Comment:
(Ab-)using the context pointer prevents clients to use `context` in their `KinoLayerCallbacks` and effectively makes the use of (future) callbacks and color inversion mutually exclusive.

Why not introducing a new field `bool invert_cololors` in `structure KinoLayer` instead?